Octokit/Clients/RepositoriesClient.cs

```diff
@@ -79,11 +81,24 @@ public Task<Repository> Create(string organizationLogin, NewRepository newReposi
             return Create(ApiUrls.OrganizationRepositories(organizationLogin), organizationLogin, newRepository);
         }
 
+        [Preview(AcceptHeaders.TemplateRepositoryPreviewCodeName)]
+        [ManualRoute("POST", "/repos/{owner}/{repo}/generate")]
+        public Task<Repository> Generate(string owner, string repo, NewRepositoryFromTemplate newRepository)
+        {
+            Ensure.ArgumentNotNull(owner, nameof(owner));
+            Ensure.ArgumentNotNull(repo, nameof(repo));
+            Ensure.ArgumentNotNull(newRepository, nameof(newRepository));
+            if (string.IsNullOrEmpty(newRepository.Name))
+                throw new ArgumentException("The new repository's name must not be null.");
+
+            return ApiConnection.Post<Repository>(ApiUrls.Repositories(owner, repo), newRepository, AcceptHeaders.TemplateRepositoryPreview);
```

Comment:
You have some failing tests in `Octokit.Tests/Clients/RepositoriesClientTests.cs` because the assertions are not set up to look for this new preview header. These need to be updated, like this:

```C#
[Fact]
public async Task TheNewRepositoryDescription()
{
    var connection = Substitute.For<IApiConnection>();
    var client = new RepositoriesClient(connection);
    var newRepository = new NewRepository("aName");

    await client.Create("aLogin", newRepository);

    connection.Received().Post<Repository>(Args.Uri, newRepository, "application/vnd.github.baptiste-preview+json");
}
```

We avoid using the `AcceptsHeader` value here as it feels like a shortcut - we want to assert the header value received matches what the API should see.